pkg/plugin/framework/server_mux.go

```diff
@@ -75,3 +82,34 @@ func (m *serverMux) getHandler(name string) (interface{}, error) {
 
 	return m.handlers[name], nil
 }
+
+// ValidatePluginName checks if the given name:
+// - the plugin name has two parts separated by '/'
+// - non of the above parts is empty
+// - the prefix is a valid DNS subdomain name,
+// - a plugin with the same name does not already exist (if list of existing names is passed in)
+func ValidatePluginName(name string, existingNames []string) error {
+	// validate there is one "/" and two parts
+	parts := strings.Split(name, "/")
+	if len(parts) != 2 {
+		return errors.Errorf("%s is not a DNS subdomain prefix followed by '/' and a non-empty name (e.g. 'example.com/name'", name)
```

Comment:
missing closing parenthesis in the error msg - should be `return errors.Errorf("%s is not a DNS subdomain prefix followed by '/' and a non-empty name (e.g. 'example.com/name')", name)`